### PR TITLE
[NTOS:MM] MiCheckForUserStackOverflow(): Improve comments and logs

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Mostly addendum to 8a04950.

Please, check I understood this function correctly.

### CODE REVIEW

File: ntoskrnl/mm/ARM3/pagfault.c [hunk 1]
```diff
@@ -78,16 +78,13 @@ MiCheckForUserStackOverflow(IN PVOID Address,
     /* This is where the stack will start now */
     NextStackAddress = (PVOID)((ULONG_PTR)PAGE_ALIGN(Address) - GuranteedSize);
 
-    /* Do we have at least one page between here and the end of the stack? */
-    if (((ULONG_PTR)NextStackAddress - PAGE_SIZE) <= (ULONG_PTR)DeallocationStack)
+    /* Is not there enough additional memory between here and the end of the stack? */
+    if (((ULONG_PTR)NextStackAddress - GuranteedSize) <= (ULONG_PTR)DeallocationStack)
```
Comment (encodedpr) [inline ntoskrnl/mm/ARM3/pagfault.c]: this is more than a comment change

**Comment by andrew-boyarshin** [comment]
This will conflict with #1070.

**Comment by SergeGautherie** [comment]
> This will conflict with #1070.

Yes, Yet trivial to unbitrot.

File: ntoskrnl/mm/ARM3/pagfault.c (hunk 1 above)
Comment (Mouvedia) [reply ntoskrnl/mm/ARM3/pagfault.c]: https://github.com/reactos/reactos/blob/c4e273e359584bcf092f9a07ecd10853926722b9/ntoskrnl/mm/ARM3/pagfault.c#L36
typo GuranteedSize > GuaranteedSize

Comment (andrew-boyarshin) [reply ntoskrnl/mm/ARM3/pagfault.c]: It's variable name, not a typo. See #1070.

Comment (learn-more) [reply ntoskrnl/mm/ARM3/pagfault.c]: This is not an improvement for the comment, you change a well formed sentence in something that is not a well formed sentence.

**Comment by HBelusca** [comment]
Please update your PR.

File: ntoskrnl/mm/ARM3/pagfault.c (hunk 1 above)
Comment (SergeGautherie) [reply ntoskrnl/mm/ARM3/pagfault.c]: > this is more than a comment change.. not saying its correct or incorrect

Not really: it is a trivial copypasta fix, as currently `GuaranteedSize = PAGE_SIZE;`.

> This is not an improvement for the comment

Partly reverted.
NB: Simply suggest a better wording, when my English is not good enough.

**Comment by SergeGautherie** [comment]
Unbitrotted.

File: ntoskrnl/mm/ARM3/pagfault.c [hunk 2]
```diff
@@ -123,12 +125,13 @@ MiCheckForUserStackOverflow(IN PVOID Address,
...
         return STATUS_PAGE_FAULT_GUARD_PAGE;
     }
 
-    /* Fail, we couldn't move the guard page */
-    DPRINT1("Guard page failure: %lx\n", Status);
+    /* Fail, we couldn't set the guard page */
+    DPRINT1("ZwAllocateVirtualMemory failed for guard page: %lx, stack end=0x%p, guarantee=%lx, next=0x%p\n",
+            Status, DeallocationStack, GuaranteedSize, NextStackAddress);
     ASSERT(FALSE);
     return STATUS_STACK_OVERFLOW;
```
Comment (SergeGautherie) [inline ntoskrnl/mm/ARM3/pagfault.c]: While here, you may suggest an `ASSERTMSG()` text...

File: ntoskrnl/mm/ARM3/pagfault.c [hunk 3]
```diff
@@ -78,16 +78,15 @@ MiCheckForUserStackOverflow(IN PVOID Address,
     /* This is where the stack will start now */
     NextStackAddress = (PVOID)((ULONG_PTR)PAGE_ALIGN(Address) - GuaranteedSize);
 
-    /* Do we have at least one page between here and the end of the stack? */
-    if (((ULONG_PTR)NextStackAddress - PAGE_SIZE) <= (ULONG_PTR)DeallocationStack)
+    /* Do we have enough additional memory between here and the end of the stack? */
+    if (((ULONG_PTR)NextStackAddress - GuaranteedSize) <= (ULONG_PTR)DeallocationStack)
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply ntoskrnl/mm/ARM3/pagfault.c]: Comment, code or both? Why? How do you want it fixed instead?

[the remark this answers is not in the document]
Comment (HeisSpiter) [reply ntoskrnl/mm/ARM3/pagfault.c]: Code

[the remark this answers is not in the document]
Comment (SergeGautherie) [reply ntoskrnl/mm/ARM3/pagfault.c]: Split off to a future PR...

**Comment by Extravert-ir** [comment]
Squash your commits and this can be merged

**Comment by SergeGautherie** [comment]
Squashed.

**Comment by HeisSpiter** [comment]
I don't see the point of such pull request. It's basically just nitpicking on comments/debug prints.
And, in the end, I find its changes less clear and its comments less precise that the ones we already had.
I'm in favour of just closing it.

**Comment by HeisSpiter** [comment]
Considering no one else responded, I take my position as representative amongst the kernel developers, and thus, closing the PR.

**Pull request closed without merging**